debugutils/aggregator.go

```diff
@@ -0,0 +1,129 @@
+package debugutils
+
+import (
+	"io"
+	"os"
+
+	"github.com/solo-io/go-utils/installutils/helmchart"
+	"github.com/solo-io/go-utils/tarutils"
+	"github.com/spf13/afero"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ZipStorageClient interface {
+	Save(fileName string, data io.Reader) error
+	Read(fileName string) (io.ReadCloser, error)
+}
+
+type LocalZipStorageClient struct {
+	fs afero.Fs
+}
+
+func NewLocalZipStorageClient(fs afero.Fs) *LocalZipStorageClient {
+	return &LocalZipStorageClient{fs: fs}
+}
+
+func (lsc *LocalZipStorageClient) Save(fileName string, data io.Reader) error {
+	file, err := lsc.fs.OpenFile(fileName, os.O_CREATE|os.O_RDWR, 0777)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+	_, err = io.Copy(file, data)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (lsc *LocalZipStorageClient) Read(file string) (io.ReadCloser, error) {
+	return lsc.fs.OpenFile(file, os.O_RDWR, 0777)
+}
+
+const (
+	aggregatorName = "aggregator"
+)
+
+type Aggregator struct {
+	col ResourceCollector
+	pf  PodFinder
+	lsc *LogStorageClient
```

Comment:
I don't get why we need two different storage clients here - seems like a strange abstraction

Reply:
Actually, in addition to this there is storage logic in the ResourceCollector and LogRequestBuilder too -- I think we should revisit all the storage-related logic to simplify the abstraction and use a consistent abstraction across the ResourceCollector and log fetcher